JustSaying/Fluent/SubscriptionsBuilder.cs

```diff
@@ -0,0 +1,134 @@
+        /// </exception>
+        public SubscriptionsBuilder ForQueue<T>()
+            where T : Message
+        {
+            return ForQueue<T>((p) => p.WithDefaultQueue());
+        }
+
+        /// <summary>
+        /// Configures a queue subscription.
+        /// </summary>
+        /// <param name="configure">A delegate to a method to use to configure a queue subscription.</param>
+        /// <returns>
+        /// The current <see cref="SubscriptionsBuilder"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="configure"/> is <see langword="null"/>.
+        /// </exception>
+        public SubscriptionsBuilder ForQueue<T>(Action<QueueSubscriptionBuilder<T>> configure)
+            where T : Message
+        {
+            if (configure == null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            var builder = new QueueSubscriptionBuilder<T>();
+
+            configure(builder);
+
+            Subscriptions.Add(builder);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Configures a topic subscription for the default topic name.
+        /// </summary>
+        /// <returns>
+        /// The current <see cref="SubscriptionsBuilder"/>.
+        /// </returns>
+        public SubscriptionsBuilder ForTopic<T>()
+            where T : Message
+        {
+            return ForTopic<T>((p) => p.IntoDefaultTopic());
+        }
+
+        /// <summary>
+        /// Configures a topic subscription.
+        /// </summary>
+        /// <param name="configure">A delegate to a method to use to configure a topic subscription.</param>
+        /// <returns>
+        /// The current <see cref="SubscriptionsBuilder"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="configure"/> is <see langword="null"/>.
+        /// </exception>
+        public SubscriptionsBuilder ForTopic<T>(Action<TopicSubscriptionBuilder<T>> configure)
+            where T : Message
+        {
+            if (configure == null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            var builder = new TopicSubscriptionBuilder<T>();
+
+            configure(builder);
+
+            Subscriptions.Add(builder);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the subscriptions for the <see cref="JustSayingFluently"/>.
+        /// </summary>
+        /// <param name="bus">The <see cref="JustSayingFluently"/> to configure subscriptions for.</param>
+        /// <exception cref="InvalidOperationException">
+        /// No instance of <see cref="IHandlerResolver"/> could be resolved.
+        /// </exception>
+        internal void Configure(JustSayingFluently bus)
+        {
+            IHandlerResolver resolver = Parent.ServiceResolver.ResolveService<IHandlerResolver>();
```

Comment:
I'm being super picky but should this be `var` or should we change line 103 to declare the type instead?